src/node_sockaddr.cc

```diff
@@ -215,6 +215,9 @@ bool in_network_ipv4(
     const SocketAddress& ip,
     const SocketAddress& net,
     int prefix) {
+  if (prefix == 32)
+    return compare_ipv4(ip, net) == SocketAddress::CompareResult::SAME;
+
   uint32_t mask = ((1 << prefix) - 1) << (32 - prefix);
```

Comment:
Not that your fix is wrong but the mask calculation has undefined behavior (signed out-of-range shift) for prefix > 30 so a more comprehensive fix is this:
```suggestion
  uint32_t mask = ((1ull << prefix) - 1) << (32 - prefix);
```
There's another instance around line 296.

prefix arguably shouldn't be an int but e.g. a uint8_t because what does a less-than-zero prefix even mean? That's a bigger change though.

Reply:
While I agree with your observation, currently there is a check around prefixes < 0 or > 32:

```
$ ~/node/bin/node 
> const bl = new net.BlockList()
undefined
> bl.addSubnet('1.1.0.0', -1, 'ipv4')
Uncaught:
RangeError [ERR_OUT_OF_RANGE]: The value of "prefix" is out of range. It must be >= 0 && <= 32. Received -1
    at __node_internal_captureLargerStackTrace (node:internal/errors:477:5)
    at new NodeError (node:internal/errors:388:5)
    at __node_internal_ (node:internal/validators:92:13)
    at BlockList.addSubnet (node:internal/blocklist:107:9) {
  code: 'ERR_OUT_OF_RANGE'
}
```

If you still feel that this is required, please let me know.

Reply:
That's right, and there are also a couple of additional C++ checks that enforce the range but still, switching to `uint8_t` would help code clarity. That's just a suggestion though, not a hard requirement for this PR to land.

The UB should be fixed though. It's dangerous.

Reply:
Apologies for the delay on this. What is the recommended approach to add the UB fix on top, should be a separate commit or should the previous commit be amended with the fix?

Reply:
Doesn't matter, what you prefer. It gets squashed into a single commit before merge anyway.